src/workos.ts

```diff
@@ -100,7 +105,18 @@ export class WorkOS {
             throw new NotFoundException(path, requestID);
           }
           default: {
-            throw new GenericServerException(status, data.message, requestID);
+            if (error && errorDescription) {
```

Comment:
Since we allow for `error` and `error_description` to be `undefined`, we should be able to collapse two of these conditions into one:

```ts
if (error || errorDescription) {
  throw new OauthException(
    status,
    requestID,
    error,
    errorDescription
  );
} else {
  throw new GenericServerException(status, data.message, requestID);
}
```

Reply:
It looks like there's a couple of spots where we'll need to do this.